### DISCUSSION TITLE
Multi-tenancy Directus v9

### DISCUSSION THREAD

**Discussion by vukvucic** [directus/directus#6383] [Archive]
Hi,

We have got Directus 9 running on an AWS Web Server and it is working perfectly. However we are looking to setup some level of multi-tenancy to allow us to use the single web-server for multiple clients as the requirements are quite low.

We have read a lot of the discussion here and it seems like we can set-up multiple instances of Directus on the webserver the only issue we were running into is how you deploy and run all of them.

We can only imagine that you would need a custom start script that would cd into every folder and run them? This seems a bit painful as we would also have to deploy every instance anytime we make a change to any one instance.

Are we missing something or is this the expected practice?

**Comment by philleepflorence** [comment]
Not adept at AWS, but see if these help: 

- [AWS Installation Guide: Directus Docs](https://docs.directus.io/guides/installation/aws/)
- [Automatically Deploy from GitHub Using AWS CodeDeploy](https://aws.amazon.com/blogs/devops/automatically-deploy-from-github-using-aws-codedeploy/)
- [PM2 Deployment](https://pm2.keymetrics.io/docs/usage/deployment/)

**Reply by vukvucic** [reply]
Yeah, we are pretty familiar with AWS deployments and things just gets a bit messy if you have multiple Directus instances on the same web server. Will look further thanks!

**Comment by benhaynes** [comment]
Have you looked into docker/kubernetes? Those might be a good option for fleet provisioning.

**Reply by philleepflorence** [reply]
@benhaynes - are these related?
- [Kubernetes: Open Source Container Orchestration System](https://kubernetes.io)
- [Fleet: Manage Large Fleets of Kubernetes Clusters](https://github.com/rancher/fleet)

**Reply by benhaynes** [reply]
Seem to be! But this is not my area of focus...

**Reply by philleepflorence** [reply]
Me neither lol...
May I add **DevOps** as a label?

**Reply by vukvucic** [reply]
Will check these out thanks!
